colossalai/pipeline/schedule/zero_bubble_pp.py

```diff
@@ -480,22 +480,26 @@ def backward_b_step(
 
         # For chunk 0 stage 0, use micro_batch as input_obj_
         if model_chunk_id == 0 and self.stage_manager.is_first_stage(ignore_chunk=True):
-            input_obj_, _ = tree_flatten(micro_batch)
-            output_obj_, _ = tree_flatten(output_obj)  # y
-            output_obj_grad_, _ = tree_flatten(output_obj_grad)  # dy
+            input_obj_, _ = tree_flatten({k: v for k, v in micro_batch.items() if isinstance(v, torch.Tensor)})
+            output_obj_, _ = tree_flatten({k: v for k, v in output_obj.items() if isinstance(v, torch.Tensor)})  # y
+            output_obj_grad_, _ = tree_flatten(
+                {k: v for k, v in output_obj_grad.items() if isinstance(v, torch.Tensor)}
+            )  # dy
```

Comment:
don't traverse such obj like dict, just use tree flatten and traverse the list

Reply:
> don't traverse such obj like dict, just use tree flatten and traverse the list

Fixed in https://github.com/hpcaitech/ColossalAI/pull/6069/commits/83163fa70c49085b15ea063fcb0ee188d28f4871；